dall-e-shell.el

```diff
@@ -419,6 +421,34 @@ ERROR-CALLBACK otherwise."
                                        "KEY-NOT-FOUND")))))
                 "-d" (shell-maker--json-encode request-data))))
 
+(defun dall-e-shell-generate-image (prompt text)
+  "Generate image from text using DALL-E given prompt."
+  (let* ((question (concat prompt text))
+         (png-output (dall-e-shell-post-prompt question)))
+    png-output))
+
+(defun dall-e-shell-generate-image-from-region-0 (prompt)
+  "Generate image from region using DALL-E given prompt."
+  (let* ((text (buffer-substring-no-properties (region-beginning) (region-end)))
```

Comment:
We should error out if there's no active region.

```emacs-lisp
  (unless (region-active-p)
    (user-error "No active region"))
```

Reply:
Thanks. Good to know.